### PR TITLE
Add change time tracking request

### PR DESCRIPTION
**Pull request by tjb7**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -28,6 +29,7 @@ export { Absence, AbsenceApprove, AbsenceCreate, AbsenceReject, AbsenceUpdate }
 export { AbsenceType } from './absenceTypes/types';
 export { Credentials } from './authentication/types';
 export { Tokens } from './baseApi';
+export { ChangeTimeTrackingRequestEndpoint } from './changeTimeTrackingRequests';
```
Comment (tomi-timetac) [inline src/index.ts:32]: Double imports?

File: src/changeTimeTrackingRequests/changeTimeTrackingRequest.test.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    const record = createMock<Resource>();
+    const apiResponse = { Success: false, NumResults: 1, Results: [record] };
+
+    mock.onGet(readPath).reply(400, apiResponse);
+    const actual: Promise<Resource[]> = endpoint.read();
+    expect(await actual.catch((err: { message: string }) => err.message)).toMatch('Request failed with status code 400');
+  });
+});
+
+describe('tasks.readRaw', () => {
```
Comment (tomi-timetac) [inline src/changeTimeTrackingRequests/changeTimeTrackingRequest.test.ts:68]: looks like duplicate?

File: src/changeTimeTrackingRequests/changeTimeTrackingRequest.test.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+import { ErrorReason, TimeTacApiError } from '../errors';
+import { RequestParamsBuilder } from '../utils/params/requestParams';
+import { ReadRawResponse } from '../utils/response/readRawResponse';
+import { ChangeTimeTrackingRequestEndpoint } from './index';
+import { ChangeTimeTrackingRequest } from './types';
+
+type Resource = ChangeTimeTrackingRequest;
+const endpoint: ChangeTimeTrackingRequestEndpoint = new ChangeTimeTrackingRequestEndpoint({ account: 'testingAccount' });
+
+describe('tasks.read', () => {
```
Comment (tomi-timetac) [inline src/changeTimeTrackingRequests/changeTimeTrackingRequest.test.ts:15]: wrong name.

**Review by tomi-timetac — CHANGES_REQUESTED** [review]

File: src/changeTimeTrackingRequests/changeTimeTrackingRequest.test.ts [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+import { RequestParamsBuilder } from '../utils/params/requestParams';
+import { ReadRawResponse } from '../utils/response/readRawResponse';
+import { ChangeTimeTrackingRequestEndpoint } from './index';
+import { ChangeTimeTrackingRequest } from './types';
+
+type Resource = ChangeTimeTrackingRequest;
+const endpoint: ChangeTimeTrackingRequestEndpoint = new ChangeTimeTrackingRequestEndpoint({ account: 'testingAccount' });
+
+describe('changeTimeTrackingRequest.read', () => {
+  const readPath = `${endpoint.getResourcePath()}/read`;
```
Comment (tjb7) [inline src/changeTimeTrackingRequests/changeTimeTrackingRequest.test.ts:16]: @tomi-timetac fixed naming

File: src/changeTimeTrackingRequests/changeTimeTrackingRequest.test.ts [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+    const apiResponse = { Success: false, NumResults: 1, Results: [record] };
+
+    mock.onGet(readPath).reply(400, apiResponse);
+    const actual: Promise<Resource[]> = endpoint.read();
+    expect(await actual.catch((err: { message: string }) => err.message)).toMatch('Request failed with status code 400');
+  });
+});
+
+describe('changeTimeTrackingRequestRead.readRaw', () => {
+  const readPath = `${endpoint.getResourcePath()}/read`;
```
Comment (tjb7) [inline src/changeTimeTrackingRequests/changeTimeTrackingRequest.test.ts:69]: @tomi-timetac I think they are differenct, read vs. readRaw, I am not sure. @whosshell could you comment if both tests are needed?

File: src/index.ts [hunk 6]
```diff
@@ -28,6 +29,7 @@ export { Absence, AbsenceApprove, AbsenceCreate, AbsenceReject, AbsenceUpdate }
 export { AbsenceType } from './absenceTypes/types';
 export { Credentials } from './authentication/types';
 export { Tokens } from './baseApi';
+export { ChangeTimeTrackingRequest } from './changeTimeTrackingRequests/types';
 export { DeltaSyncResults } from './deltaSync/types';
```
Comment (tjb7) [inline src/index.ts:33]: @tomi-timetac fixed the export, before I exported EndPoint, but should have just exported the type.

File: src/changeTimeTrackingRequests/changeTimeTrackingRequest.test.ts (hunk 5 above)
Comment (whosshell) [reply src/changeTimeTrackingRequests/changeTimeTrackingRequest.test.ts:69]: I would leave it for now. To have more clear answer I would have to check what is `epicStrategyBuilder` doing.
If more information is needed contact me directly.

**Review by whosshell — APPROVED** [review]

File: src/changeTimeTrackingRequests/changeTimeTrackingRequest.test.ts (hunk 3 above)
Comment (tomi-timetac) [reply src/changeTimeTrackingRequests/changeTimeTrackingRequest.test.ts]: still wrong name.

**Review by tomi-timetac — APPROVED** [review]

**Pull request merged**